src/main/java/com/mraof/minestuck/client/gui/DialogueScreen.java

```diff
@@ -0,0 +1,204 @@
+package com.mraof.minestuck.client.gui;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mraof.minestuck.MinestuckConfig;
+import com.mraof.minestuck.entity.consort.DialogueCard;
+import com.mraof.minestuck.network.DialogueOptionPacket;
+import com.mraof.minestuck.network.MSPacketHandler;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.widget.button.Button;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraftforge.fml.client.gui.widget.ExtendedButton;
+import org.lwjgl.glfw.GLFW;
+
+import java.util.List;
+
+public class DialogueScreen extends Screen
+{
+	private static final ResourceLocation BACKGROUND = new ResourceLocation("minestuck", "textures/gui/dialogue.png");
+	
+	private static final int GUI_WIDTH = 256;
+	private static final int GUI_HEIGHT = 106;
+	private static final int PORTRAIT_SIZE = 32;
+	
+	private final List<DialogueCard> dialogueCards;
+	private final List<String> responseOptions;
+	
+	private String renderedText;
+	private int currentCardIndex;
+	private boolean doneWriting;
+	private int frame;
+	
+	public DialogueScreen(List<DialogueCard> dialogueCards, List<String> responseOptions)
+	{
+		super(new TranslationTextComponent("minestuck.dialogue"));
+		this.dialogueCards = dialogueCards;
+		this.currentCardIndex = 0;
+		this.responseOptions = responseOptions;
+		resetWriter();
+	}
+	
+	@Override
+	public void render(int mouseX, int mouseY, float partialTicks)
+	{
+		this.renderBackground();
+		
+		int xOffset = (width - GUI_WIDTH) / 2;
+		int yOffset = (height - GUI_HEIGHT) / 2;
+		int leftStart = xOffset + PORTRAIT_SIZE + 16;
+		
+		RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
+		this.minecraft.getTextureManager().bindTexture(BACKGROUND);
+		this.blit(xOffset, yOffset, 0, 0, GUI_WIDTH, GUI_HEIGHT);
```

Comment:
The bottom section(black box with yellow/red outline) of the dialogue is cut off, is this intentional?

Reply:
As well, it is important to note that the light gray background crashes with the lighter color texts of salamanders and iguanas. Something such as an alternate backdrop color should be done to provide more contrast

Reply:
I wanted to keep consistency for the backgrounds so I tweaked the colors a bit but its not perfect. I can add it back if its something we want.

Reply:
For the dialogue section, I would have to make it scalable and I removed it for simplicity to concentrate on the integration